A2017/tensorflow.ipynb

```diff
@@ -0,0 +1,1357 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Attempting to use uninitialized value v\n",
+      "\t [[Node: _retval_v_0_0 = _Retval[T=DT_FLOAT, index=0, _device=\"/job:localhost/replica:0/task:0/device:CPU:0\"](v)]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    try:\n",
+    "        session.run(v)\n",
+    "    except tf.errors.FailedPreconditionError as e:\n",
+    "        print(e)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "TensorFlow provides a function, `tf.global_variables_initializer`, which returns an op that can be evaluated to do just that:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The value for v is [ 0.  0.]\n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    session.run(tf.global_variables_initializer())\n",
+    "    print('The value for v is {}'.format(session.run(v)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A variable's value persists across a session unless it is updated by running an assignment op. For instance, the op returned by `tf.assign_add` can be used to increment a variable's value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The value for v is [ 0.  0.]\n",
+      "The value for v is [ 0.  0.]\n",
+      "The value for v is [ 1.  2.]\n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    session.run(tf.global_variables_initializer())\n",
+    "    # The value for v persists across session.run calls...\n",
+    "    print('The value for v is {}'.format(session.run(v)))\n",
+    "    print('The value for v is {}'.format(session.run(v)))\n",
+    "    # ... until it is updated by running an assignment op.\n",
+    "    session.run(v.assign_add([1, 2]))\n",
+    "    print('The value for v is {}'.format(session.run(v)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In addition to `tf.assign_add`, the `tf.assign_sub` and `tf.assign` functions return ops which decrement and assign a variable's value, respectively.\n",
+    "\n",
+    "## Placeholder\n",
+    "\n",
+    "Oftentimes the computation we define depends on data which we don't yet have. For instance, the output of a neural network depends on a user-defined input which is only specified at runtime.\n",
```

Comment:
Maybe mention (even orally) that it is for when those values are computed outside of tensorflow

Reply:
Will do.